### PR TITLE
[Portworx] Add idempotency for local snapshots

### PR DESCRIPTION
**Pull request by harsh-px**
Signed-off-by: Harsh Desai <harsh@portworx.com>

**What type of PR is this?**
>bug

**What this PR does / why we need it**:

If stork crashes after invoking the driver to create a snapshot but before returning success back to the snapshot provisioner, then the snapshot is created in the backend. But stork will keep retrying to bind the snapshot and will fail since the snapshot with the same ID already exists.

**Does this PR change a user-facing CRD or CLI?**:
no

**Is a release note needed?**:
```release-note
Add idempotency to local snapshot calls to handle crashes while a snapshot is being taken.
```

**Does this change need to be cherry-picked to a release branch?**:
no

### CODE REVIEW

**Comment by disrani-px** [comment]
Add a `[Portworx]` or `Portworx` prefix to the commit message since this is a driver only change.

**Comment by harsh-px** [comment]
> Add a [Portworx] or Portworx prefix to the commit message since this is a driver only change.


Done

File: drivers/volume/portworx/portworx.go [hunk 1]
```diff
@@ -844,7 +844,23 @@ func (p *portworx) SnapshotCreate(
 		}
 		snapshotID, err = volDriver.Snapshot(volumeID, true, locator, true)
 		if err != nil {
-			return nil, getErrorSnapshotConditions(err), err
+			// Check already exists error and return existing snapshot if found
+			if isAlreadyExistsError(err) {
```
Comment (disrani-px) [inline drivers/volume/portworx/portworx.go]: File a bug to return a proper typed error from `volDriver.Snapshot()`.

Comment (harsh-px) [reply drivers/volume/portworx/portworx.go]: Done PWX-8959

**Pull request merged**